docs/src/model_simulation/examples/periodic_events_simulation.md

```diff
@@ -0,0 +1,95 @@
+```@example periodic_event_example
+using OrdinaryDiffEq, Plots
+u0 = [:X => 150.0, :Xᴾ => 50.0]
+ps = [:kₚ => 0.1, :kᵢ => 0.1, :l => 1.0]
+oprob = ODEProblem(circadian_model, u0, (0.0, 100.0), ps)
+sol = solve(oprob)
+plot(sol)
+```
+
+## [Modelling a circadian periodic event in a jump simulation](@id periodic_event_simulation_example_ode)
+While defining periodic events is easy for ODE and SDE simulations, due to how events are internally implemented, these cannot currently be used for jump simulations. Instead, we create a new model (and its corresponding `JumpProblem`), without the event:
+```@example periodic_event_example
+circadian_model = @reaction_network begin
+ (kₚ*l,kᵢ), X <--> Xᴾ
+end
+
+using JumpProcesses
+u0 = [:X => 150, :Xᴾ => 50]
+ps = [:kₚ => 0.1, :kᵢ => 0.1, :l => 1.0]
+dprob = DiscreteProblem(circadian_model, u0, (0.0, 100.0), ps)
+jprob = JumpProblem(circadian_model, dprob, Direct())
+nothing # hide
+```
+Next, we implement our event through a `DiscreteCallback`. It triggers every 12 time units, and when it does, it flips the value of `l`. Since we will interface with $l$'s value multiple times, we [create specific `getl` and `setl` functions](@ref simulation_structure_interfacing_functions) to do this (in practice, due to this model's small size, the performance impact is negliable).
+```@example periodic_event_example
+getl = ModelingToolkit.getp(jprob, :l)
+setl = ModelingToolkit.setp(jprob, :l)
+function condition(u, t, integrator)
+ (t % 12) == 0.0
+end
+function affect!(integrator)
+    setl(integrator, (getl(integrator) + 1) % 2)
+    reset_aggregated_jumps!(integrator)
+end
+callback = DiscreteCallback(condition, affect!)
+nothing # hide
+```
+!!! danger
+    Whenever the integrator of a jump simulator is modified, the `reset_aggregated_jumps!` function must be called (with the integrator as its input) afterwards. Omitting to do so will result in erroneous simulations.
+
+Next, if we simulate our model, we note that the events do not seem to be triggered
+```@example periodic_event_example
+sol = solve(jprob, SSAStepper(); callback)
+plot(sol)
+```
+The reason is that discrete callbacks' conditions are only checked at the end of each simulation time step, and the probability that these will coincide with the callback's trigger times ($t = 12, 24, 36, ...$) is infinitely small. Hence, we must directly instruct our simulation to stop at these time points. We can do this using the `tstops` argument:
+```@example periodic_event_example
+tstops = 12.0:12.0:dprob.tspan[2]
+sol = solve(jprob, SSAStepper(); callback, tstops)
+plot(sol)
+```
+
+## [Plotting the light level](@id periodic_event_simulation_plotting_light)
+Sometimes when simulating models with periodic parameters, one would like to plot the parameter's value across the simulation. To do this, we must turn our parameter $l$ to a *variable* (so that its value is recorded throughout the simulation):
```

Comment:
Shouldn't this example instead use a saving callback (https://docs.sciml.ai/DiffEqCallbacks/stable/output_saving/)? That seems preferred to making the parameter into a variable when it really isn't a variable?

Reply:
I don't think there could be a plotting receipe since you can save anything via it if I understand correctly.

Reply:
right, that makes sense

Reply:
Right now I only make a reference to this, hopefully at some point I can look into it properly and make a full section.